cmd/tsconnect/wasm/wasm_js.go

```diff
@@ -465,3 +512,17 @@ func generateHostname() string {
 	scale := scales[rand.Intn(len(scales))]
 	return fmt.Sprintf("%s-%s", tail, scale)
 }
+
+// promiseWrapper handles the boilerplate of wrapping Goroutines with JS
```

Comment:
lowercase goroutines

but describe what it actually does? "makePromise returns a JS Promise that yields the result of calling f with ..." ?